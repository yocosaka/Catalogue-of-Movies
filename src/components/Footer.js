import React from 'react';
import { Link } from 'react-router-dom';

const Footer = () => (
  <footer className="footer">
    <Link to="/">
      <span className="footer__logo">
        MoviSearch
      </span>
    </Link>
    <div className="footer__copyright">
      <Link to="https://github.com/yocosaka" target="_blank" rel="noreferrer">&copy; Yoko Saka</Link>
    </div>
  </footer>
);

export default Footer;
