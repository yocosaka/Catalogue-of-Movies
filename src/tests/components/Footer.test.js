import React from 'react';
import { BrowserRouter, MemoryRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Footer from '../../components/Footer';

describe('Footer with Link', () => {
  test('full app rendering/navigating', () => {
    render(<Footer />, { wrapper: MemoryRouter });
    // eslint-disable-next-line no-unused-expressions
    expect(screen.getByText('MoviSearch')).toBeInTheDocument;
  });

  test('should match with snapshot', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Footer />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
