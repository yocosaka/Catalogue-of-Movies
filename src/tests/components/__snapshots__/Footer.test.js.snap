// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer with Link should match with snapshot 1`] = `
<footer
  className="footer"
>
  <a
    href="/"
    onClick={[Function]}
  >
    <span
      className="footer__logo"
    >
      MoviSearch
    </span>
  </a>
  <div
    className="footer__copyright"
  >
    <a
      href="/https://github.com/yocosaka"
      onClick={[Function]}
      rel="noreferrer"
      target="_blank"
    >
      Â© Yoko Saka
    </a>
  </div>
</footer>
`;
